### PR TITLE
update existing tests to use `ExpectedException` instead of `@Test(expected = SomeException.class)`

### PR DESCRIPTION
**Pull request by albertlr**
As for issue #19, I did the following changes:

* update existing tests to use `ExpectedException` instead of `@Test(expected = SomeException.class)` and 
* fix `ThrowingBinaryOperatorTest` to use `ThrowingBinaryOperator::unchecked` instead of `ThrowingBiFunction::unchecked`

@pivovarit could you take a look?

### CODE REVIEW

File: src/test/java/pl/touk/throwing/ThrowingBiConsumerTest.java [hunk 1]
```diff
@@ -51,15 +59,22 @@ public void shouldConsumeAsFunction() throws Exception {
...
     public void shouldConsumeAndThrowUnchecked() throws Exception {
+        final IOException cause = new IOException("some message");
+        
+        thrown.expect(WrappedException.class);
+        thrown.expectMessage("some message");
+        thrown.expectCause(is(cause));
+
         // given
-        ThrowingBiConsumer<Integer, Integer, IOException> consumer = (i, j) -> { throw new IOException(); };
+        ThrowingBiConsumer<Integer, Integer, IOException> consumer = (i, j) -> { throw cause; };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingBiConsumerTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingBiFunctionTest.java [hunk 2]
```diff
@@ -46,37 +54,55 @@ public void shouldReturnEmptyOptional() throws Exception {
...
 
-    @Test(expected = Exception.class)
+    @Test
     public void shouldThrowEx() throws Exception {
+        thrown.expect(Exception.class);
+        thrown.expectMessage("some message");
+
         // given
-        final ThrowingBiFunction<Integer, Integer, Integer, Exception> f1 = (i, j) -> { throw new Exception(); };
+        final ThrowingBiFunction<Integer, Integer, Integer, Exception> f1 = (i, j) -> { throw new Exception("some message"); };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingBiFunctionTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingBiFunctionTest.java [hunk 3]
```diff
@@ -46,37 +54,55 @@ public void shouldReturnEmptyOptional() throws Exception {
...
     public void shouldWrapInWrappedEx() throws Exception {
+        final Exception cause = new Exception("some message");
+        
+        thrown.expect(WrappedException.class);
+        thrown.expectMessage("some message");
+        thrown.expectCause(is(cause));
+
         // given
-        final ThrowingBiFunction<Integer, Integer, Integer, Exception> f1 = (i, j) -> { throw new Exception(); };
+        final ThrowingBiFunction<Integer, Integer, Integer, Exception> f1 = (i, j) -> { throw cause; };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingBiFunctionTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingBiFunctionTest.java [hunk 4]
```diff
@@ -46,37 +54,55 @@ public void shouldReturnEmptyOptional() throws Exception {
...
     public void shouldWrapInRuntimeExWhenUsingUnchecked() throws Exception {
+        final Exception cause = new Exception("some message");
+        
+        thrown.expect(WrappedException.class);
+        thrown.expectMessage("some message");
+        thrown.expectCause(is(cause));
+
         // given
-        final ThrowingBiFunction<Integer, Integer, Integer, Exception> f1 = (i, j) -> { throw new Exception(); };
+        final ThrowingBiFunction<Integer, Integer, Integer, Exception> f1 = (i, j) -> { throw cause; };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingBiFunctionTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingConsumerTest.java [hunk 5]
```diff
@@ -51,15 +60,22 @@ public void shouldConsumeAsFunction() throws Exception {
...
     public void shouldConsumeAndThrowUnchecked() throws Exception {
+        final IOException cause = new IOException("some message");
+        
+        thrown.expect(WrappedException.class);
+        thrown.expectMessage("some message");
+        thrown.expectCause(is(cause));
+
         // given
-        ThrowingConsumer<Integer, IOException> consumer = i -> { throw new IOException(); };
+        ThrowingConsumer<Integer, IOException> consumer = i -> { throw cause; };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingConsumerTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingRunnableTest.java [hunk 6]
```diff
@@ -1,21 +1,34 @@
...
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
     public void shouldRun() throws Exception {
+        thrown.expect(IOException.class);
+        thrown.expectMessage("some message");
+        
         // given
-        ThrowingRunnable<Exception> runnable = () -> { throw new IOException(); };
+        ThrowingRunnable<Exception> runnable = () -> { throw new IOException("some message"); };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingRunnableTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingRunnableTest.java [hunk 7]
```diff
@@ -29,26 +42,40 @@ public void shouldRunUnchecked() throws Exception {
...
     public void shouldRunUncheckedAndThrow() throws Exception {
+        final IOException cause = new IOException("some message");
+        
+        thrown.expect(WrappedException.class);
+        thrown.expectMessage("some message");
+        thrown.expectCause(is(cause));
+
         // given
-        ThrowingRunnable<Exception> runnable = () -> { throw new IOException(); };
+        ThrowingRunnable<Exception> runnable = () -> { throw cause; };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingRunnableTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingRunnableTest.java [hunk 8]
```diff
@@ -29,26 +42,40 @@ public void shouldRunUnchecked() throws Exception {
...
     public void shouldRunUncheckedAndThrowUsingUtilsMethod() throws Exception {
+        final IOException cause = new IOException("some message");
+        
+        thrown.expect(WrappedException.class);
+        thrown.expectMessage("some message");
+        thrown.expectCause(is(cause));
+
         // given
-        ThrowingRunnable<Exception> runnable = () -> { throw new IOException(); };
+        ThrowingRunnable<Exception> runnable = () -> { throw cause; };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingRunnableTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingSupplierTest.java [hunk 9]
```diff
@@ -70,26 +78,40 @@ public void shouldGetUnchecked() throws Exception {
...
     public void shouldGetUncheckedAndThrow() throws Exception {
+        final IOException cause = new IOException("some message");
+        
+        thrown.expect(WrappedException.class);
+        thrown.expectMessage("some message");
+        thrown.expectCause(is(cause));
+
         // given
-        final ThrowingSupplier<Integer, IOException> supplier = () -> { throw new IOException(); };
+        final ThrowingSupplier<Integer, IOException> supplier = () -> { throw cause; };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingSupplierTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingSupplierTest.java [hunk 10]
```diff
@@ -70,26 +78,40 @@ public void shouldGetUnchecked() throws Exception {
...
     public void shouldGetUncheckedWithUtilsFunction() throws Exception {
+        final IOException cause = new IOException("some message");
+        
+        thrown.expect(WrappedException.class);
+        thrown.expectMessage("some message");
+        thrown.expectCause(is(cause));
+
         // given
-        final ThrowingSupplier<Integer, IOException> supplier = () -> { throw new IOException(); };
+        final ThrowingSupplier<Integer, IOException> supplier = () -> { throw cause; };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingSupplierTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

File: src/test/java/pl/touk/throwing/ThrowingUnaryOperatorTest.java [hunk 11]
```diff
@@ -29,24 +38,33 @@ public void shouldApplyUnchecked() throws Exception {
...
     public void shouldApplyUncheckedAndThrow() throws Exception {
+        final IOException cause = new IOException("some message");
+        
+        thrown.expect(WrappedException.class);
+        thrown.expectMessage("some message");
+        thrown.expectCause(is(cause));
+
         // given
-        ThrowingUnaryOperator<Integer, IOException> op = i -> { throw new IOException(); };
+        ThrowingUnaryOperator<Integer, IOException> op = i -> { throw cause; };
```
Comment (sputnikci) [inline src/test/java/pl/touk/throwing/ThrowingUnaryOperatorTest.java]: \[Checkstyle\] ERROR: Only one statement per line allowed\.

**Comment by pivovarit** [comment]
@albertlr thanks for the contribution :) unfortunately this library is effectively dead. Those features were introduced in https://github.com/javaslang/javaslang. but I will be glad to have a look

**Comment by albertlr** [comment]
@pivovarit 
Well ... unfortunately `javaslang` is way too much for my needs :)
Your project is exactly what I was looking for.

**Comment by pivovarit** [comment]
Ok, that one looks fine :) thanks

**Pull request merged**